java/rocksjni/ratelimiterjni.cc

```diff
@@ -22,3 +23,98 @@ jlong Java_org_rocksdb_GenericRateLimiterConfig_newRateLimiterHandle(
       static_cast<int64_t>(jrefill_period_micros),
       static_cast<int32_t>(jfairness)));
 }
+
+/*
+ * Class:     org_rocksdb_RateLimiter
+ * Method:    newRateLimiterHandle
+ * Signature: (JJI)J
+ */
+jlong Java_org_rocksdb_RateLimiter_newRateLimiterHandle(
+    JNIEnv* env, jclass jclazz, jlong jrate_bytes_per_second,
+    jlong jrefill_period_micros, jint jfairness) {
+  auto* rate_limiter = rocksdb::NewGenericRateLimiter(
+      static_cast<int64_t>(jrate_bytes_per_second),
+      static_cast<int64_t>(jrefill_period_micros),
+      static_cast<int32_t>(jfairness));
+
+  // TODO(AR) `ptr_sptr_rate_limiter` is never deleted :-(
```

Comment:
Yes, there's no direct translation of shared_ptr between C++ and Java.  I think the best we can do is to have a pointer holding a shared_ptr (as what we did in this PR.)

In the C API, we simply create a struct that has shared_ptr instance (instead of pointer) for each shared_ptr we would like to expose.

https://github.com/facebook/rocksdb/blob/4.10.fb/db/c.cc#L106-L107

For instance, we use struct rocksdb_logger_t as a place holder of shared_ptr<Logger>.  rocksdb_logger_t here served as same purpose as the `handle` in RocksObject in our JNI.  

So I think we can safely delete the handle as it only means its associated Java RateLimiter cannot hold any more shared_ptr, and its shared ownership of the object pointed by shared_ptr is already released by handle->reset().